.github/workflows/json-validator.yaml

```diff
@@ -0,0 +1,31 @@
+name: Validate JSON Files
+
+on:
+  push:
+    paths:
+      - '**/*.json'
+  pull_request:
+    paths:
+      - '**/*.json'
+
+jobs:
+  validate-json:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install jq
+        run: sudo apt-get install jq
+
+      - name: Validate JSON
+        run: |
+          for file in $(find . -type f -name '*.json'); do
+            jq . "$file" >/dev/null || { echo "Invalid JSON: $file"; exit 1; }
+            last_char=$(tail -c 1 "$file")
```

Comment:
Not sure if this works consistently - it errors out on every file for me, even those that clearly have newlines at the end.

StackOverflow suggests something like this to detect files without a concluding newline:
`if [[ $(tail -c1 "$file" | wc -l) -eq 0 ]]`